dev/TabView/TabViewListView.cpp

```diff
@@ -40,18 +40,40 @@ void TabViewListView::OnItemsChanged(winrt::IInspectable const& item)
 {
     __super::OnItemsChanged(item);
 
-    if (auto tabView = SharedHelpers::GetAncestorOfType<winrt::TabView>(winrt::VisualTreeHelper::GetParent(*this)))
+    if (const auto tabView = SharedHelpers::GetAncestorOfType<winrt::TabView>(winrt::VisualTreeHelper::GetParent(*this)))
     {
-        auto internalTabView = winrt::get_self<TabView>(tabView);
+        const auto internalTabView = winrt::get_self<TabView>(tabView);
         internalTabView->OnItemsChanged(item);
     }
 }
 
+void TabViewListView::PrepareContainerForItemOverride(const winrt::DependencyObject& element, const winrt::IInspectable& item)
+{
+    const auto itemContainer = element.as<winrt::TabViewItem>();
+    const auto tvi = winrt::get_self<TabViewItem>(itemContainer);
+
+    // Due to virtualization, a TabViewItem might be recycled to display a different tab data item.
+    // In that case, there is no need to set the TabWidthMode of the TabViewItem or its parent TabView
+    // as they are already set correctly here.
+    //
+    // We know we are currently looking at a TabViewItem being recycled if its parent TabView has already been set.
+    if (!tvi->GetParentTabView())
```

Comment:
>GetParentTabView [](start = 14, length = 16)

Do we need to call GetAncestorOfType if we have this to get access to the TabView?

Reply:
@StephenLPeters Observe that the check is `if (tvi->GetParentTabView() == nullptr)`. We need the GetAncestorOfType() call here because we only are interested in the parent TabView when the TabViewItem has been *newly* created (and not recycled). When we are looking at a fresh TabViewItem object, it doesn't yet have its parent TabView set so we need to walk the visual tree upwards here.